### DISCUSSION TITLE
Drop .pyi?

### DISCUSSION THREAD

**Discussion by scoates** [PyGithub/PyGithub#1993] [Ideas]
Any opinions on dropping the `.pyi` definitions? I'd need to do a bit more research to be sure, but maybe they're [no longer needed](https://github.com/PyGithub/PyGithub/commit/63e4fae997a9a5dc8c2b56907c87c565537bb28f) if we can roll the hints directly into the Python?

I could spend a little time on further research + a PR for this if there's no opposition.

(The itch I'm trying to scratch is that they're already out of sync with the Python, so my IDE complains, needlessly, which technically could just be a PyCharm problem, but I think the way forward is direct. e.g. [`ProjectCard.move(…)`](https://github.com/PyGithub/PyGithub/blob/83d6e0719cb7bfee57af398f93a73516b62af5f2/github/ProjectCard.py#L138) is missing from [the type def](https://github.com/PyGithub/PyGithub/blob/83d6e0719cb7bfee57af398f93a73516b62af5f2/github/ProjectCard.pyi).)

**Comment by xmo-odoo** [comment]
An advantage of having separate pyi files, aside from the ability to quickly scan interfaces without needing an external tool (in theory the stubs could even have the docstrings though I think that would cause Sphinx some issues) is that they have a non-insignificant runtime cost (during import).

PEP 563 / 649 (deferral of annotations evaluation to mitigate this) was originally slated to become the default behaviour in 3.10, but turns out to break the world, so for now it's been postponed to 3.11, conditional of finding solutions for existing users: https://mail.python.org/archives/list/python-dev@python.org/thread/CLVXXPQ2T2LQ5MP2Y53VVQFCXYWQJHKZ/
